lib/IRGen/GenStruct.cpp

```diff
@@ -563,36 +563,29 @@ namespace {
     const clang::RecordDecl *ClangDecl;
 
     const clang::CXXConstructorDecl *findCopyConstructor() const {
-      const clang::CXXRecordDecl *cxxRecordDecl =
-          dyn_cast<clang::CXXRecordDecl>(ClangDecl);
+      const auto *cxxRecordDecl = dyn_cast<clang::CXXRecordDecl>(ClangDecl);
       if (!cxxRecordDecl)
         return nullptr;
-      for (auto method : cxxRecordDecl->methods()) {
-        if (auto ctor = dyn_cast<clang::CXXConstructorDecl>(method)) {
-          if (ctor->isCopyConstructor() &&
-              ctor->getAccess() == clang::AS_public &&
-              // rdar://106964356
-              // ctor->doesThisDeclarationHaveABody() &&
-              !ctor->isDeleted())
-            return ctor;
-        }
+      for (auto ctor : cxxRecordDecl->ctors()) {
+        if (ctor->isCopyConstructor() &&
+            ctor->getAccess() == clang::AS_public &&
+            // rdar://106964356
```

Comment:
Please don't add naked radar links to the main source code like this.  First and foremost, you should describe the issue that you're working around instead of just adding a link.  Second, this is an open-source project, so you should really link to an open source GitHub issue which will be readable by non-Apple employees.

Reply:
Got it. This one was here before my change. I'll try to replace it with the summary of what the rdar is about in a followup PR.